### PR TITLE
[AF-1432] Remove AF dependencies from Uberfire Core Grids component

### PR DESCRIPTION
**Pull request by danielezonca**
Removed all UF/WB related dependencies to Uberfire Core Grids component to become a pure Lienzo/Gwt component and maybe moved to an independent Lienzo-Grid component in the future.

- Removed `PortablePreconditions` usages
- Changed `Callback` to `Consumer`
- Changed `Command` to `Runnable`
- Changed `Pair` to `Map.Entry`

@manstis @mdproctor @romartin @jsoltes 

PRs related:
https://github.com/kiegroup/appformer/pull/425
https://github.com/kiegroup/drools-wb/pull/902
https://github.com/kiegroup/kie-wb-common/pull/2009

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/EnumMultiSelectUiColumn.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Consumer;
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/EnumMultiSelectUiColumn.java]: Not removed `Callback` import and one more `Callback` occurrence in this class.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java [hunk 2]
```diff
@@ -76,12 +76,12 @@ public ValueListUiColumn(final List<HeaderMetaData> headerMetaData,
     @Override
     public void doEdit(final GridCell<String> cell,
                        final GridBodyCellRenderContext context,
-                       final Callback<GridCellValue<String>> callback) {
+                       final Consumer<GridCellValue<String>> callback) {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java]: Noticed in all files, we now have variables `Consumer callback`, should we refactor to `Consumer consumer`? Not sure, just asking.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/EnumMultiSelectUiColumn.java (hunk 1 above)
Comment (danielezonca) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/EnumMultiSelectUiColumn.java]: Sorry probably PR description is a bit misleading (I did it for `appformer` module and then copy/paste in all the related PRs).
The scope of the PR is to update only UberFire Wires Core Grids component with those changes and then update all the corresponding usages. It is not intended to change all `Callback` with `Consumer` in the WB codebase.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java (hunk 2 above)
Comment (danielezonca) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java]: See comment above

**Pull request merged**